src/Resolver/Location/FindParentStrategy.php

```diff
@@ -138,4 +159,50 @@ public function updateParent(ElementInterface $element, array $inputData): Eleme
     protected function loadById()
     {
     }
+
+    /**
+     * @throws InvalidInputException
+     */
+    private function setElementType(ElementInterface $element, DataObject | ElementInterface $newParent): void
+    {
+        // Check if element should be saved as a variant if not already.
+        if (
+            $this->saveAsVariant && $element->getType() !== AbstractObject::OBJECT_TYPE_VARIANT
```

Comment:
another case that came into my mind.
I have a variant, I cannot change the type to object anymore. The other way round it is possible. 
I would say, either we rename the option 'force variant', or we allow changing both ways (with the implemented restrictions).

Reply:
@turbo-ele :ping_pong:

Reply:
you are right, i would say it should be changeable in both ways. i'm going to change it

Reply:
@turbo-ele can you have a look at phpstan.